index.js

```diff
@@ -1,97 +1,226 @@
 var fs = require('fs');
-var http = require('http');
-var https = require('https');
 var request = require('request');
-var emoji = require('node-emoji')
+var moment = require('moment');
 
-const {
-    IncomingWebhook
-} = require('@slack/client');
-const url = process.env.SLACK_WEBHOOK_URL;
+const UrlStatus = {
+    Up: 'Up',
+    Pending: 'Pending',
+    StillDown: 'StillDown',
+};
 
-const webhookurl = "https://hooks.slack.com/services/<some-sort-of-secret-key>";
-const webhook = new IncomingWebhook(webhookurl);
-var notReachable = {};
-var notReachabletwice = {};
+Object.freeze(UrlStatus);
 
+const webhookUrl = process.env.SLACK_WEBHOOK_URL;
+const urlContainer = {};
+const urlErrorContainer = {};
 
 /* Read configuration and start loops */
 fs.readFile('./config.json', 'utf8', function (err, data) {
     if (err) throw err;
-    var conf = JSON.parse(data);
-    var urls = conf.urls;
+    const conf = JSON.parse(data);
+    const urls = conf.urls;
 
-    for (var name in urls) {
-        var url = urls[name].url;
-        var interval = urls[name].interval;
-        createloop(name, url, interval);
+    for (const [name, params] of Object.entries(urls)) {
+        createLoop(name, params);
     }
 });
 
 /* Create a loop that's executed every x milliseconds */
-function createloop(name, url, interval) {
-    var requestLoop = setInterval(function () {
-        request({
-            url: url,
-            method: "GET",
+function createLoop(name, params) {
+    setInterval(function () {
+        const requestDate = new Date();
+        const dateBounds = {
+            minTimeOfDay: params.minTimeOfDay,
+            maxTimeOfDay: params.maxTimeOfDay,
+            daysOfTheWeek: params.daysOfTheWeek
+        }
+
+        if (!isDateWithinBounds(requestDate, dateBounds)) {
+            urlContainer[name] = undefined;
+            return;
+        }
+
+        const defaultRequestConfig = {
             timeout: 10000,
-            followRedirect: true,
-            maxRedirects: 10
-        }, function (error, response, body) {
-            if (!error && response.statusCode == 200) {
-                if (typeof notReachable[name] != "undefined") {
-                  if (notReachable[name] == "notagain") {
-                    delete notReachable[name];
-                    var message = name + " " + emoji.get('beers') + " " + getFormattedDate();
-                    send(message);
-                    console.log(message);
-                  }
-                }
+            followRedirect: true
+        };
+
+        const requestConfigWithUrl = Object.assign(
+            { url: params.url },
+            params.requestConfig ?? defaultRequestConfig
```

Comment:
no need to use `Object.assign()`, go with a simple spread, and always spread `defaultRequestConfig`:

```js
const requestConfig = {
   ...defaultRequestConfig,
   ...params.requestConfig, // if undefined, it will still work
   url: params.url,
};
```